packages/datadog-plugin-jest/src/index.js

```diff
@@ -191,9 +192,14 @@ module.exports = [
   {
     name: 'jest-environment-node',
     versions: ['>=24.8.0'],
-    patch: function (NodeEnvironment, tracer) {
+    patch: function (NodeEnvironment, tracer, config) {
       const testEnvironmentMetadata = getTestEnvironmentMetadata('jest')
-
+      // it if is the default service, we substitute it
+      const serviceUnderTest = findPkg('package.json', process.cwd()).name
```

Comment:
What makes using `findPkg` here than in `pkg.js`? Does it return a different value? Could this be done directly in `pkg.js` instead? The main reason for executing the function directly in `pkg.js` is that the service name should only be detected once, and it should be the same regardless of where `pkg` is used. It also makes it (or at least should make it) possible to test that the right service name is returned for an "external" app defined in tests.

Reply:
I exported `findPkg` so that you can pass a different `cwd` (`findRoot`). The function as it was defaulted to `require.main.filename` (in [here](https://github.com/DataDog/dd-trace-js/blob/d1c1c1ef6e171a01500842b21f29639ea0a08dcc/packages/dd-trace/src/pkg.js#L7-L8), which is *not* what we want for testing instrumentations, since the "application under test" is somewhere else. 

What I can do is create a different file specifically for testing instrumentations? The thing is that the logic will be very similar, as we're also looking for a `package.json`

Reply:
This is not unique to testing instrumentations. The reason this happens is because it's a parent module that is the root of the process, but this could happen in applications as well when some sort of CLI runner is used, for example with `next`, `micro`, etc. If we can make this logic generic so that it works with not only Jest and the likes, but also with any runner, that would fix the issue completely.

Reply:
wouldn't a `findPkg` that receives a "root" solve this? That was my reasoning behind this change. With this we let each package handle where they want to find their "root"

Reply:
The instrumentation doesn't matter. For example, if for a Jest process the root is not Jest but the actual service under test, that is true for the entire tracer, not just the `jest` integration.